File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/05.While-Loop/03-additional/02.reportSystem.js
function reportSystem(input) {

    const expectedAmount = Number(input[0]);
    let index = 1;
    let current = input[index++];
    let counterPayments = 0;
    let totalAmount = 0;
    let moneyOnHand = 0;
    let counterMoneyOnHand = 0;
    let card = 0;
    let counterCard = 0;

    while (current !== "End") {
        let currentProductPrice = Number(current);
        counterPayments++;
        // По условие се редуват първо в брой плащане, след това чрез кредитна карта.
        if (counterPayments % 2 !== 0) {
            // • Ако продуктът надвишава 100лв., за него не може да се плати в брой.
            if (currentProductPrice > 100) {
                console.log("Error in transaction!");
            } else {
                totalAmount += currentProductPrice;
                moneyOnHand += currentProductPrice;
                counterMoneyOnHand++;
                console.log("Product sold!");
            }
        } else {
            // • Ако продуктът е на цена под 10лв., за него не може да се плати с кредитна карта.
            if (currentProductPrice < 10) {
                console.log("Error in transaction!");
            } else {
                totalAmount += currentProductPrice;
                card += currentProductPrice;
                counterCard++;
                console.log("Product sold!");
            }
        }

        if (totalAmount >= expectedAmount) {
            let averageCS = moneyOnHand / counterMoneyOnHand;
            let averageCC = card / counterCard;
            console.log(`Average CS: ${averageCS.toFixed(2)}`);
            console.log(`Average CC: ${averageCC.toFixed(2)}`);
            break;
        }

        current = input[index++];
    }

    if (current === "End") {
        console.log("Failed to collect required money for charity.");
    }
}
reportSystem(["600", "86", "150", "98", "227", "End"]);

File: Python_Basics/Homeworks-And-Labs/05.while_loop/03.additional/02.report_system.py
expected_amount = int(input())
current = input()
counter_payments = 0
total_amount = 0
money_on_hand = 0
counter_money_on_hand = 0
card = 0
counter_card = 0

while current != "End":
    current_product_price = int(current)
    counter_payments += 1
    # По условие се редуват първо в брой плащане, след това чрез кредитна карта.
    if counter_payments % 2 != 0:
        # Ако продуктът надвишава 100лв., за него не може да се плати в брой.
        if current_product_price > 100:
            print("Error in transaction!")
        else:
            total_amount += current_product_price
            money_on_hand += current_product_price
            counter_money_on_hand += 1
            print("Product sold!")
    else:
        # Ако продуктът е на цена под 10лв., за него не може да се плати с кредитна карта.
        if current_product_price < 10:
            print("Error in transaction!")
        else:
            total_amount += current_product_price
            card += current_product_price
            counter_card += 1
            print("Product sold!")
            
    if total_amount >= expected_amount:
        averageCS = money_on_hand / counter_money_on_hand
        averageCC = card / counter_card
        print(f'Average CS: {averageCS:.2f}')
        print(f'Average CC: {averageCC:.2f}')
        break
        
    current = input()
    
if current == "End":
    print("Failed to collect required money for charity.")
